tools/build_documentation_release.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# Script for building docs
+# It builds the latest docs and pulls the historical stable docs
+# from the git repository
+set -e
+
+# Build the documentation.
+tox -edocs
+
+# Extract the release version from Cargo.toml
+VERSION=$(grep -Po -m1 'version = "\K[0-9]+\.[0-9]+' Cargo.toml)
+
+TMP_DIR=$(mktemp -d)
+git clone --depth 1 --branch gh-pages https://github.com/Qiskit/rustworkx.git $TMP_DIR
+rm -rf $TMP_DIR/stable/$VERSION # Remove the old version in case of a revision release
+mkdir -p $TMP_DIR/stable/$VERSION
+cp -r docs/build/html/* $TMP_DIR/stable/$VERSION/
+mkdir -p release_docs
+cp -r $TMP_DIR/* release_docs/
```

Comment:
Doesn't this mean the root of the docs becomes the dev build and we only see the released documentation on `stable/$VERSION`? The current hosting has the root documentation as the current release, `/dev` for the dev builds and `/stable/*` for all the historical builds (including the current release).

Reply:
That is correct, dev would be at the root. I can try changing it to stable at the root later but it gets complicated. Especially because we have no sidebar button to go to the dev version

Reply:
I am updating the code to mimic the old behaviour, it shouldn't be too hard. I will put it on hold temporarily